docs/dev/compiler-toolchain/vyper.md

```diff
@@ -0,0 +1,74 @@
+# Vyper
+
+The compiler we provide as a part of our toolchain is called [zkvyper](https://github.com/matter-labs/zkvyper-bin). It
+operates on LLL IR and metadata received from the underlying [vyper](https://docs.vyperlang.org/en/latest/index.html) compiler,
+which must be available in `$PATH`, or the path must be explicitly passed via the CLI (command-line interface).
+
+## Usage
+
+Most of the time, using our compiler via the hardhat plugin is enough, and you do not have to dive deeper into its
+interface and I/O (input/output) methods. However, it can be crucial for integration, debugging, or contribution purposes.
+
+#### Combined JSON
+
+The hardhat plugin uses the combined JSON I/O mode, which is passed down to the `vyper` compiler child process.
+
+Additional zkEVM data is inserted into the output combined JSON by `zksolc`:
+
+- `zk_version`: the version of the `zksolc` itself
+- `contract/factory_deps`: bytecode hashes of contracts `CREATE`d in the current one.
```

Comment:
```suggestion
- `contract/factory_deps`: bytecode hashes of contracts; currently `CREATE`d.
```

Reply:
'Currently `CREATE`d' loses the information that the contracts must be created in the contract the `factory_deps` field belongs to.
Also it may have a shade of perfect tense, but the contracts are actually `CREATE`d **after** the deployment of the project you're compiling, that is during executing transactions on it.

Reply:
oh, i had no idea you were making up a word ... making another suggestion ...